model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/lock/entity/JpaLockEntity.java

```diff
@@ -0,0 +1,169 @@
+
+    //used for implicit optimistic locking
+    @Version
+    @Column
+    private int version;
+
+    @Type(type = "jsonb")
+    @Column(columnDefinition = "jsonb")
+    private final JpaLockMetadata metadata;
+
+    @Column(insertable = false, updatable = false)
+    @Basic(fetch = FetchType.LAZY)
+    private Integer entityVersion;
+
+    @Column(insertable = false, updatable = false)
+    @Basic(fetch = FetchType.LAZY)
+    private String name;
+
+    /**
+     * No-argument constructor, used by hibernate to instantiate entities.
+     */
+    public JpaLockEntity() {
+        this.metadata = new JpaLockMetadata();
+    }
+
+    public JpaLockEntity(DeepCloner cloner) {
+        this.metadata = new JpaLockMetadata(cloner);
+    }
+
+    public JpaLockEntity(final UUID id, final int version, final Integer entityVersion, String name) {
+        this.id = id;
+        this.version = version;
+        this.entityVersion = entityVersion;
+        this.name = name;
+        this.metadata = null;
+    }
+
+    public boolean isMetadataInitialized() {
+        return metadata != null;
+    }
+
+    @Override
+    public Integer getEntityVersion() {
+        if (isMetadataInitialized()) return metadata.getEntityVersion();
+        return entityVersion;
+    }
+
+    @Override
+    public Integer getCurrentSchemaVersion() {
+        return CURRENT_SCHEMA_VERSION_GROUP;
+    }
+
+    @Override
+    public void setEntityVersion(Integer entityVersion) {
+        metadata.setEntityVersion(entityVersion);
+    }
+
+    @Override
+    public int getVersion() {
+        return version;
+    }
+
+    @Override
+    public String getId() {
+        return id == null ? null : id.toString();
+    }
+
+    @Override
+    public void setId(String id) {
+        String validatedId = UuidValidator.validateAndConvert(id);
+        this.id = UUID.fromString(validatedId);
+    }
+
+    public String getName() {
+        if (isMetadataInitialized()) return metadata.getName();
+        return name;
+    }
+
+    @Override
+    public void setName(String name) {
+        metadata.setName(name);
+    }
+
+    @Override
+    public String getKeycloakInstanceIdentifier() {
+        return metadata.getKeycloakInstanceIdentifier();
+    }
+
+    @Override
+    public void setKeycloakInstanceIdentifier(String keycloakInstanceIdentifier) {
+        metadata.setKeycloakInstanceIdentifier(keycloakInstanceIdentifier);
+    }
+
+    @Override
+    public Long getTimeAcquired() {
+        return metadata.getTimeAcquired();
+    }
+
+    @Override
+    public void setTimeAcquired(Long timeAcquired) {
+        metadata.setTimeAcquired(timeAcquired);
+    }
+
+    @Override
+    public int hashCode() {
+        return getClass().hashCode();
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) return true;
+        if (!(obj instanceof JpaGroupEntity)) return false;
+        return Objects.equals(getId(), ((JpaGroupEntity) obj).getId());
+    }
```

Comment:
```suggestion
    public boolean equals(Object obj) {
        if (this == obj) return true;
        if (!(obj instanceof JpaLockEntity)) return false;
        return Objects.equals(getId(), ((JpaLockEntity) obj).getId());
    }
```

Reply:
Oops, well spotted, will be fixed in next commit